routing/router.go

```diff
@@ -538,6 +546,30 @@ func (r *ChannelRouter) Start() error {
 		return err
 	}
 
+	// Before we restart existing payments and start accepting more
+	// payments to be made, we clean the network result store of the
+	// Switch. We do this here at startup to ensure no more payments can be
+	// made concurrently, so we know the toKeep map will be up-to-date
+	// until the cleaning has finished.
+	toKeep := make(map[uint64]struct{})
+	for _, p := range payments {
+		payment, err := r.cfg.Control.FetchPayment(
+			p.Info.PaymentHash,
+		)
+		if err != nil {
+			return err
+		}
+
+		for _, a := range payment.HTLCs {
```

Comment:
It seems to me that `p` which is an `InFlightPayment` has an `Attempts` member which holds the same info? (Could be I'm wrong here lacking deeper knowledge of the underlying code.)

Reply:
I deleted that field in the first commit, since it was not used for anything else :)

Reply:
Need to upgrade my memory to be able the handle three consecutive commits :)